File: 0206: reverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur = head
        lst = []
        while cur != None:
            lst.append(cur.val)
            cur = cur.next
        lst = lst[::-1]
        count = len(lst)
        cur = head
        i = 0
        while cur != None:
            cur.val = lst[i] 
            cur = cur.next
            i = i +1
        return head

File: 0206: reverseLinkedList.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    
    int size = 0;
    struct ListNode* cur = head;
    while(cur != NULL){

        size++;
        cur = cur->next;
    }

    cur = head;
    int* arr = (int*)malloc(size*sizeof(int));
    int i = 0;
    while(cur != NULL){

        arr[i] = cur->val;
        i++;
        cur = cur->next;
    }

    cur = head;
    for(int i = 0; i < size; i++){

        cur->val = arr[size-1-i];
        cur = cur->next;
    }
    
    return head;
}